React/Base/RCTUtils.m

```diff
@@ -347,6 +347,7 @@ BOOL RCTClassOverridesInstanceMethod(Class cls, SEL selector)
   if (error) {
     errorMessage = error.localizedDescription ?: @"Unknown error from a native module";
     errorInfo[@"domain"] = error.domain ?: RCTErrorDomain;
+    errorInfo[@"userInfo"] = error.userInfo;
```

Comment:
There's also no guarantee that error.userInfo is not nil, and assigning nil to a dictionary will crash on iOS7 IIRC. So this should probably be:
  
  errorInfo[@"userInfo"] = RCTNullIfNil(RCTJSONClean(error.userInfo));